samples/msal-angular-v2-samples/angular10-browser-sample/src/app/app.component.ts

```diff
@@ -1,26 +1,40 @@
-import { Component, OnInit, Inject } from '@angular/core';
-import { MsalService } from './msal';
-import { MSAL_GUARD_CONFIG, InteractionType } from './msal/constants';
+import { Component, OnInit, Inject, OnDestroy } from '@angular/core';
+import { MsalService, MsalBroadcastService } from './msal';
+import { MSAL_GUARD_CONFIG, InteractionType, MsalBroadcastEvent } from './msal/constants';
 import { MsalGuardConfiguration } from './msal/msal.guard.config';
+import { Subject } from 'rxjs';
+import { filter, takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, OnDestroy {
   title = 'Angular 10 - MSAL Browser Sample';
   isIframe = false;
   loggedIn = false;
+  private readonly _destroying$ = new Subject<void>();
 
   constructor(
     @Inject(MSAL_GUARD_CONFIG) private msalGuardConfig: MsalGuardConfiguration,
-    private authService: MsalService
+    private authService: MsalService,
+    private broadcastService: MsalBroadcastService
   ) {}
+
   ngOnInit(): void {
     this.isIframe = window !== window.parent && !window.opener;
 
     this.checkAccount();
+
+    this.broadcastService.msalSubject$
+      .pipe(
+        filter(msg => msg.type === MsalBroadcastEvent.LOGIN_SUCCESS || msg.type === MsalBroadcastEvent.ACQUIRE_TOKEN_SUCCESS),
```

Comment:
Don't have to do this here, but in the library we might consider providing convenience functions that do this filtering for each event type.

Reply:
Noted!

Reply:
I don't know how fancy you want to get here, but I often go the type-safe-enum-ish route with stuff like this.

In constants.ts:
```ts
export enum MsalBroadcastEventType {
    LOGIN_SUCCESS = "msal:loginSuccess",
    LOGIN_FAILURE = "msal:loginFailure",
    ACQUIRE_TOKEN_SUCCESS = "msal:acquireTokenSuccess",
    ACQUIRE_TOKEN_FAILURE = "msal:acquireTokenFailure",
    SSO_SILENT_SUCCESS = "msal:ssoSilentSuccess",
    SSO_SILENT_FAILURE = "msal:ssoSilentFailure"
}

export interface MsalBroadcastEvent {
    eventType: MsalBroadcastEventType;
    accountChanged: boolean;
    // ...whatever else comes to mind, can be added to as needed
}

export const msalBroadcastEvents: { [key: string]: MsalBroadcastEvent } = {
    loginSucess: {
        eventType: MsalBroadcastEventType.LOGIN_SUCCESS,
        accountChanged: true
    },
    loginFailure: {
        eventType: MsalBroadcastEventType.LOGIN_FAILURE,
        accountChanged: false, // (would this be false?)
    },
    acquireTokenSuccess: {
        eventType: MsalBroadcastEventType.ACQUIRE_TOKEN_SUCCESS,
        accountChanged: true
    },
    // ...
};
```

All that ceremony just so you can do this in AppComponent:
```ts
this.broadcastService.msalSubject$
    .pipe(
        filter(msg => msg.type.accountChanged),
        takeUntil(this._destroying$)
    )
    .subscribe(msg => {
        this.checkAccount();
    });
```
I haven't given the names much thought. Convenience functions might well be easier anyway.